forest/jasmine/traj2stats.py

```diff
@@ -819,121 +824,290 @@ def gps_quality_check(study_folder: str, study_id: str) -> float:
+        tz_str, string, timezone
+        frequency, Frequency, the frequency of the summary stats
+            (resolution for summary statistics)
+        save_traj, bool, True if you want to save the trajectories as a
+            csv file, False if you don't
+        places_of_interest: list of amenities or leisure places to watch,
+            keywords as used in openstreetmaps
+        save_log, bool, True if you want to output a log of locations
+            visited and their tags
+        threshold, int, time spent in a pause needs to exceed the
+            threshold to be placed in the log
+            only if save_log True, in minutes
+        split_day_night, bool, True if you want to split all metrics to
+            datetime and nighttime patterns
+            only for daily frequency
+        person_point_radius, float, radius of the person's circle when
+            discovering places near him in pauses
+        place_point_radius, float, radius of place's circle
+            when place is returned as centre coordinates from osm
+        time_start, time_end are starting time and ending time of the
+            window of interest
+            time should be a list of integers with format
+            [year, month, day, hour, minute, second]
+            if time_start is None and time_end is None: then it reads all
+            the available files
+            if time_start is None and time_end is given, then it reads all
+            the files before the given time
+            if time_start is given and time_end is None, then it reads all
+            the files after the given time
+        beiwe_id: a list of beiwe IDs
+        parameters: hyperparameters in functions, recommend to set it to
+            none (by default)
+        all_memory_dict and all_bv_set are dictionaries from previous run
+            (none if it's the first time)
+    Returns:
+        write summary stats as csv for each user during the specified
+            period
+        and a log of all locations visited as a json file if required
+        and imputed trajectory if required
+        and memory objects (all_memory_dict and all_bv_set)
+            as pickle files for future use
+        and a record csv file to show which users are processed
+        and logger csv file to show warnings and bugs during the run
     """
-    
-    quality_threshold = quality_threshold if quality_threshold is not None else 0.05
-    
-    if os.path.exists(output_folder)==False:
-        os.mkdir(output_folder)
 
-    if parameters == None:
-        parameters = [60*60*24*10,60*60*24*30,0.002,200,5,1,0.3,0.2,0.5,100,0.01,0.05,3,10,2,'GLC',10,51,None,None,None]
-    [l1,l2,l3,g,a1,a2,b1,b2,b3,d,sigma2,tol,switch,num,linearity,method,itrvl,accuracylim,r,w,h] = parameters
-    pars0 = [l1,l2,l3,a1,a2,b1,b2,b3]
-    pars1 = [l1,l2,a1,a2,b1,b2,b3,g]
+    if not os.path.exists(output_folder):
+        os.mkdir(output_folder)
 
-    if r == None:
+    if parameters is None:
```

Comment:
Would it make sense to make `parameters` a dataclass?
https://docs.python.org/3.8/library/dataclasses.html